core/src/main/java/org/apache/iceberg/hadoop/HadoopFileIO.java

```diff
@@ -211,6 +212,19 @@ private ExecutorService executorService() {
     return executorService;
   }
 
+  private void deletePath(FileSystem fs, Path toDelete, boolean recursive) throws IOException {
+    Trash trash = new Trash(fs, getConf());
```

Comment:
I'm concerned about number of Trash objects we create. Does Hadoop API ensure we can reuse the trash object for a given (fs, conf)? 
I couldn't tell from https://hadoop.apache.org/docs/stable/api/org/apache/hadoop/fs/Trash.html#Trash-org.apache.hadoop.fs.FileSystem-org.apache.hadoop.conf.Configuration-